git/git.go

```diff
@@ -163,8 +163,24 @@ func Log(sha1, sha2 string) (string, error) {
 	return outputs, nil
 }
 
-func Remotes() ([]string, error) {
-	return gitOutput("remote", "-v")
+func Remotes() (map[string]string, error) {
+	names, err := gitOutput("remote")
+	if err != nil {
+		return nil, fmt.Errorf("couldn't list remote names: %v", err)
+	}
+
+	remotes := make(map[string]string)
+	for _, name := range names {
+		out, err := gitOutput("ls-remote", "--get-url", name)
+		if err != nil {
+			return nil, fmt.Errorf("couldn't get url for remote %s: %v", name, err)
+		}
+		if len(out) != 1 {
```

Comment:
Sorry, I wanted to add an error message for the return which would have made the intention here clear, but I got distracted.

This is an assertion or a defense to ensure that git-ls-remote --get-url always returns one and only one URL, in the unlikely event that this changes.